src/framework/dockwindow/qml/Muse/Dock/DockFrame.qml

```diff
@@ -113,10 +113,10 @@ Rectangle {
         draggingTabsAllowed: root.hasTabBar && !root.hasSingleTab
 
         tabBarCpp: Boolean(root.frameCpp) ? root.frameCpp.tabWidget.tabBar : null
-        tabsModel: frameModel.tabs
+        tabsModel: frameModel.tabsModel
         currentIndex: Boolean(root.frameCpp) && root.frameCpp.currentIndex >= 0 ? root.frameCpp.currentIndex : 0
 
-        currentToolbarComponent: frameModel.currentDockToolbarComponent
+        currentToolbarComponent: Boolean(tabsModel) ? tabsModel.data(tabsModel.index(currentIndex, 0), DockTabsModel.ToolBarComponentRole) : null
```

Comment:
Null check shouldn't be necessary; frameModel is instantiated here and referenced by id, so never null, and tabsModel is directly initialised in the constructor of frameModel, so never null either.